File: Easy/001.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map(); // Stores number → index

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (map.has(complement)) {
            return [map.get(complement), i]; // Found the pair
        }
        map.set(nums[i], i); // Store the current number with its index
    }

    return []; // If no solution found (won't happen per problem guarantee)
};

File: Easy/001.cpp
// 1. Two Sum 
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen; // maps number -> index

        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];

            // Check if complement already exists in map
            if (seen.find(complement) != seen.end()) {
                return {seen[complement], i};
            }

            // Store the current number with its index
            seen[nums[i]] = i;
        }

        return {};
    }
};